sdks/python/apache_beam/io/avroio_test.py

```diff
@@ -149,17 +151,37 @@ def _run_avro_test(
   def test_schema_read_write(self):
     with tempfile.TemporaryDirectory() as tmp_dirname:
       path = os.path.join(tmp_dirname, 'tmp_filename')
-      rows = [beam.Row(a=1, b=['x', 'y']), beam.Row(a=2, b=['t', 'u'])]
+      rows = [beam.Row(a=-1, b=['x', 'y']), beam.Row(a=2, b=['t', 'u'])]
       stable_repr = lambda row: json.dumps(row._asdict())
       with TestPipeline() as p:
         _ = p | Create(rows) | avroio.WriteToAvro(path) | beam.Map(print)
       with TestPipeline() as p:
         readback = (
             p
             | avroio.ReadFromAvro(path + '*', as_rows=True)
+            | SqlTransform("SELECT * FROM PCOLLECTION")
             | beam.Map(stable_repr))
         assert_that(readback, equal_to([stable_repr(r) for r in rows]))
 
+  def test_avro_schema_to_beam_schema_with_nullable_atomic_fields(self):
+    with tempfile.TemporaryDirectory() as tmp_dirname_input:
+      input_path = os.path.join(tmp_dirname_input, 'tmp_filename.avro')
+      parsed_schema = fastavro.parse_schema(json.loads(self.SCHEMA_STRING))
+      with open(input_path, 'wb') as tmp_avro_file:
+        fastavro.writer(tmp_avro_file, parsed_schema, self.RECORDS)
+
+      with tempfile.TemporaryDirectory() as tmp_dirname_output:
+
+        with TestPipeline() as p:
+          _ = (
+              p
+              | avroio.ReadFromAvro(input_path, as_rows=True)
+              | SqlTransform("SELECT * FROM PCOLLECTION")
+              | avroio.WriteToAvro(tmp_dirname_output))
+        with TestPipeline() as p:
+          readback = (p | avroio.ReadFromAvro(tmp_dirname_output + "*"))
+          assert_that(readback, equal_to(RECORDS))
```

Comment:
Heads up I think RECORDS doesn't actually have any records with null values. May be good to include them if we're testing nullability conversion here